pinot-plugins/pinot-input-format/pinot-avro-base/src/main/java/org/apache/pinot/plugin/inputformat/avro/AvroRecordExtractorV2.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.plugin.inputformat.avro;
+
+import org.apache.avro.generic.GenericData;
+
+
+/**
+ * Fixes the issue with extracting Avro's Fixed type.
+ */
+public class AvroRecordExtractorV2 extends AvroRecordExtractor {
```

Comment:
We don't need a V2 for this. Let's directly add it to `AvroRecordExtractor`

Reply:
`Fixed.toString()` doesn't really make lots of sense, e.g. `[1, 10, 15]`

Reply:
@Jackie-Jiang Right, it doesn't make sense to use that. My only worry is that some production tables are already using string representation. Fixing this would break their applications.

Reply:
There's no easy way to check if a string column comes from an avro field of type Fixed.

Reply:
Adding V2 will be very hard to manage and use. If the column is configured as `STRING`, with the fix it will be converted to hex-encoded string which is fine. I'm pretty sure no one rely on this wrong behavior, so strongly suggest directly add it into the default extractor.

Reply:
We can add a config for AvroRecordExtractor like `serialize_fixed_type` with default value of false. If the config is true, we use string representation, if false, we use byte[]. This way we can set the config for existing tables as true. For new tables, default value of false always uses byte[]. 
Having said that, I agree with you that most likely no one is using the string representation of Fixed types. Let me talk to my team tomorrow and I'll get back to you here.